tests/rbd/test_rbd_pwl_cache_cluster_operation.py

```diff
@@ -0,0 +1,419 @@
+        rados_obj (RadosOrchestrator): The RadosOrchestrator instance.
+        pwl (PersistentWriteAheadLog): The PersistentWriteAheadLog instance.
+        config (dict): The test configuration.
+        image_spec (str): The image specification for testing.
+    """
+    log.info("Test to restart OSD service along with pwl cache test")
+
+    def restart_osd():
+        cmd = "ceph orch ls osd"
+        out = cache_client.exec_command(cmd=cmd, output=True)
+
+        # Regular expression to extract the first OSD name
+        osd_daemon = re.search(r"osd\.(\S+)", str(out))
+        cmd = f"ceph orch restart {osd_daemon[0]}"
+        cache_client.exec_command(cmd=cmd, output=True)
+
+    with parallel() as p:
+        p.spawn(restart_osd)
+        p.spawn(verify_cache, pwl, config, cache_client, image_spec)
+
+
+def restart_ceph_target_with_cache_test(
+    admin_node, pwl, config, cache_client, image_spec
+):
+    """
+    Restart Ceph target service and verify the persistent write cache.
+
+    Args:
+        admin_node (Node): The admin node for restarting the ceph.target service.
+        pwl (PersistentWriteAheadLog): The PersistentWriteAheadLog instance.
+        config (dict): The test configuration.
+        cache_client (Node): The client node for cache testing.
+        image_spec (str): The image specification for testing.
+    """
+    log.info("Test to restart ceph target service along with pwl cache test")
+
+    def restart_ceph_target():
+        cmd = "systemctl restart ceph.target"
+        admin_node.exec_command(cmd=cmd, sudo=True)
+
+    with parallel() as p:
+        p.spawn(restart_ceph_target)
+        p.spawn(verify_cache, pwl, config, cache_client, image_spec)
+
+
+def remove_ceph_service(service_obj, host, cache_client, pwl, config, image_spec):
+    """
+    Remove a Ceph service (e.g., Mon) from a host and verify the
+    persistent write cache in parallel.
+
+    Args:
+        service_obj: The Ceph service object (e.g., Mon, OSD).
+        host (str): The name of the host where the service will be removed.
+        cache_client (Node): The client node for cache testing.
+        pwl (PersistentWriteAheadLog): The PersistentWriteAheadLog instance.
+        config (dict, optional): The test configuration. Default is None.
+        image_spec (str): The image specification for testing.
+    """
+    log.info(
+        f"Test to remove ceph {service_obj.daemon_type} service along with pwl cache test"
+    )
+
+    def remove_service():
+        cmd = f"ceph orch host label rm {host} {service_obj.daemon_type}"
+        cache_client.exec_command(cmd=cmd, output=True)
+
+        if not service_obj.remove_mon_service(host=host):
+            log.error(f"Could not remove the {service_obj.daemon_type} host")
+            return 1
+        log.info(f"{service_obj.daemon_type} removed successfully")
+
+    with parallel() as p:
+        p.spawn(remove_ceph_service)
```

Comment:
Are you calling remove_ceph_service inside remove_ceph_service method? Is this intentional?

Reply:
Also This implementation still doesn't catch any return value and throw exception right? As we discussed.